src/test/java/io/strimzi/kafka/bridge/http/HttpBridgeTestBase.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2016, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.kafka.bridge.http;
+
+import io.strimzi.kafka.bridge.KafkaClusterTestBase;
+import io.strimzi.kafka.bridge.amqp.AmqpConfig;
+import io.strimzi.kafka.bridge.config.BridgeConfig;
+import io.strimzi.kafka.bridge.config.KafkaConfig;
+import io.strimzi.kafka.bridge.config.KafkaConsumerConfig;
+import io.strimzi.kafka.bridge.http.services.BaseService;
+import io.strimzi.kafka.bridge.http.services.ConsumerService;
+import io.strimzi.kafka.bridge.http.services.ProducerService;
+import io.strimzi.kafka.bridge.http.services.SeekService;
+import io.strimzi.kafka.bridge.utils.Urls;
+import io.vertx.core.Vertx;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@ExtendWith(VertxExtension.class)
+@SuppressWarnings({"checkstyle:JavaNCSS"})
+class HttpBridgeTestBase extends KafkaClusterTestBase {
+
+    private static final Logger log = LoggerFactory.getLogger(HttpBridgeTestBase.class);
+
+    static Map<String, Object> config = new HashMap<>();
+
+    static {
+        config.put(AmqpConfig.AMQP_ENABLED, true);
+        config.put(KafkaConfig.KAFKA_CONFIG_PREFIX + ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        config.put(KafkaConsumerConfig.KAFKA_CONSUMER_CONFIG_PREFIX + ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+    }
+
+    // for periodic/multiple messages test
+    static final int PERIODIC_MAX_MESSAGE = 10;
+    static final int PERIODIC_DELAY = 200;
+    static final int MULTIPLE_MAX_MESSAGE = 10;
+    static final int TEST_TIMEOUT = 60;
+    int count;
+
+    // for request configuration
+    private static final long RESPONSE_TIMEOUT = 2000L;
+
+    Vertx vertx;
+    HttpBridge httpBridge;
+    WebClient client;
+
+    BridgeConfig bridgeConfig;
+
+    BaseService baseService() {
+        return new BaseService(client);
+    }
+
+    ConsumerService consumerService() {
+        return new ConsumerService(client);
+    }
+
+    SeekService seekService() {
+        return new SeekService(client);
+    }
+
+    ProducerService producerService() {
+        return new ProducerService(client);
+    }
```

Comment:
about the above `xxxService` methods.
Every time you call one of this (and you do that even more times during a single test) you are creating a new `xxxService` instance and always with the same client.
The simpler solution could be calling just one time an `xxxService' method in each test and save a reference to the returned object to avoid creating a new one later.
The other solution, which could be better imho, could be having this `xxxService` as singleton?

Reply:
Yes, sure, having singleton is a good idea here

Reply:
After trying to implement singleton here I found blocker for this approach... we initialize `WebClient` instance for every test method in `before` method and because of the rule that xxxService should use `WebClient` we need to recreate it for every new method. @ppatierno Can we move code from `before` method to static block?

Reply:
Looks like moving `before` to static block is not possible because of having code ``` vertx.deployVerticle(this.httpBridge, context.succeeding(id -> context.completeNow()));``` where we use `VertxTestContext`
@ppatierno any other ideas?

Reply:
But I don't think that we need a new WebClient for each test. Even that client can be a singleton :-)

Reply:
But, the client uses `Vertx` during initialization ```WebClient.create(vertx, new WebClientOptions()...``` which completes after the execution of test method. Or I don't understand the hint.

Reply:
so the WebClient depends on Vert.x instance which is create as new instance before each test which is something we don't really need.
We should try to create the Vert.x instance once for all the tests (in a @BeforeAll method)).
In this way I think all the services and the WebClient can be singleton.
The httpbridge creation and the verticle deployment should stay in the `HttpBridgeTestBase.before` method (it's possible that we want to change something on the bridge for each test). Wdyt? Is it feasible?

Reply:
@ppatierno Finally, singleton was implemented, look the last commit plz

Reply:
Good to know! You mentioned that you had a lot of failures initializing Vert.x instance just one time at the beginning of all tests.